spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/sampler/UuidToBigIntegerConverterTest.java

```diff
@@ -0,0 +1,52 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import lombok.Data;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.math.BigInteger;
+import java.util.UUID;
+
+import static java.math.BigInteger.ONE;
+import static java.math.BigInteger.ZERO;
+import static junitparams.JUnitParamsRunner.$;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.springframework.cloud.sleuth.sampler.UuidToBigIntegerConverterTest.TestData.sample;
+
+@RunWith(JUnitParamsRunner.class)
+public class UuidToBigIntegerConverterTest {
+
+	private static final BigInteger TWO = new BigInteger("2");
```

Comment:
food for thought..

junit is always new instance for each test. If you run with the normal `@RunWith(Parameterized.class)`, you get what might end up as less clutter.

```
@Parameterized.Parameter(0)
public long mostSignificantBits;
@Parameterized.Parameter(1)
public long leastSignificantBits;
...
```

IOTW, you don't need to create a TestData type